File: 백계환/입문/section1/lec03/Lec03Main.java
package section1.lec03;

import section1.lec02.Person;

public class Lec03Main {
    public static void main(String[] args) {

    }

    public static void printAgeIfPerson(Object obj) {
        if (obj instanceof Person) {
            Person person = (Person) obj;
            System.out.println(person.getAge());
        }
    }
}

File: 백계환/입문/section1/lec03/Lec03Main.kt
package lec03

import section1.lec03.Person


fun main() {
    val number1 = 3
    val number2: Long = number1.toLong()

    // 변수가 nullable이라면 적절한 처리가 필요하다.
    val number3: Int? = 3
    val number4: Long = number3?.toLong() ?: 0L

    printAgeIfPerson(Person("", 26))
    printAgeIfPerson(null)

    val person = Person("백계환", 100)
    println("이름: ${person.name}")

    val str = """
        ABC
        DEF
        ${person.age}
    """.trimIndent()
    print(str)
}

fun printAgeIfPerson(obj: Any?) {
    if (obj is Person) {
        println(obj.age) // 스마트캐스트
    }
    val person = obj as? Person
    println(person?.age)
}
